Config as actual YAML in Pipeline resource

vixus0 opened the pull request:
This change replaces `pipelineString` in the `Pipeline` resource with
`config` which expects a YAML object which represents a Concourse
pipeline configuration.

Commit:
Config as actual YAML in Pipeline resource

This change replaces `pipelineString` in the `Pipeline` resource with
`config` which expects a YAML object which represents a Concourse
pipeline configuration.

chrisfarms commented:
awesome ... both myself and rafal have attempted to sort this and failed at various hurdles.

Few things:

1. Can we support BOTH `pipelineString` and `config` at first ... otherwise this is gonna break everyone's pipelines as soon as it is deployed.... if we can support both we can transition more easily without scaring people
1. I'm 99% sure this should break `make generate` ... since controller-gen will cry about not being able to generate the type because atc.Config contains things like `map[string]interface{}` which I don't think kubebuilder pre 2.0.0 can handle .... I suspect you worked about this by _not_ using the generation at all.... are we ok with this? ... if so maybe we should remove the "`/ +k8s:openapi-gen=true` annotation on this type to prevent it attempting to overwrite the crd

chrisfarms commented:
awesome ... both myself and rafal have attempted to sort this and failed at various hurdles.

Few things:

1. Can we support BOTH `pipelineString` and `config` at first ... otherwise this is gonna break everyone's pipelines as soon as it is deployed.... if we can support both we can transition more easily without scaring people
1. I'm 99% sure this should break `make generate` ... since controller-gen will cry about not being able to generate the type because atc.Config contains things like `map[string]interface{}` which I don't think kubebuilder pre 2.0.0 can handle .... I suspect you worked about this by _not_ using the generation at all.... are we ok with this? ... if so maybe we should remove the "`/ +k8s:openapi-gen=true` annotation on this type to prevent it attempting to overwrite the crd

vixus0 commented:
Yeah I only ran this through `make docker-build` which I guess doesn't run the generate stuff? I can have it fallback to `pipelineString` if `config` isn't available.

chrisfarms commented:
no doesn't look like the docker build re-generates anything ... I think it's ok to manage the CRD manually if this works, but we should remove the generate annotations from the `Pipeline` type to avoid it breaking next time we need to run `make manifests` or `make generate`

vixus0 commented:
Will do

Commit:
Fallback to pipelineString if config not defined

We want to remain backwards-compatible with Pipeline resources defined
with `pipelineString`.

vixus0 commented:
Taking out the genclient annotations in `pipeline_types.go` caused `make docker-build` to start failing so I've kept them in there :thinking:
